src/utils/getHeaderCellClassModes.ts

```diff
@@ -1,10 +1,11 @@
 import type {NoStrictEntityMods} from '@bem-react/classname';
-import type {Header} from '@tanstack/react-table';
+
+import type {Header} from '../types/tanstack';
 
 import {getColumnPinningClassModes} from './getColumnPinningClassModes';
 
-export const getHeaderCellClassModes = <TData>(
-    header: Header<TData, unknown>,
+export const getHeaderCellClassModes = <TData, TValue = unknown>(
+    header: Header<TData, TValue>,
```

Comment:
I guess that there's something wrong with our extension for `ColumnDef`, because typescript can't calculate correct type when we use `unknown` here instead of generic's parameter type, and looks like it throws some parts of the base union away:

Property 'accessorFn' is missing in type 'ColumnDefBase<TData, any> & StringHeaderIdentifier' but required in type 'AccessorFnColumnDefBase<TData, any>'

Reply:
upd: this also reproduces on the current version of gravity-ui/table in user-side code